### PR TITLE
create shopping cart service

### PR DESCRIPTION
**Pull request by ConfusedSatlan**

### CODE REVIEW

File: src/main/java/mate/academy/service/ShoppingCartService.java [hunk 1]
```diff
@@ -0,0 +1,20 @@
+package mate.academy.service;
+
+import mate.academy.model.MovieSession;
+import mate.academy.model.ShoppingCart;
+import mate.academy.model.User;
+
+public interface ShoppingCartService {
+    /**
```
Comment (andrii-hoienko) [inline src/main/java/mate/academy/service/ShoppingCartService.java:8]: Please add documentation for all methods, or remove it everywhere

File: src/main/java/mate/academy/model/ShoppingCart.java [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+    private User user;
+
+    @OneToMany(fetch = FetchType.LAZY)
+    @JoinColumn(name = "shopping_cart_id")
+    private List<Ticket> tickets;
+
+    public ShoppingCart() {
+    }
+
+    public ShoppingCart(User user) {
```
Comment (andrii-hoienko) [inline src/main/java/mate/academy/model/ShoppingCart.java:34]: dont create methods you dont use

**Review by andrii-hoienko — CHANGES_REQUESTED** [review]
Good, but let`s improve your solution

**Review by Denis-Balako — CHANGES_REQUESTED** [review]
Get this hw from the platform. Need to fix a build.

**Review by Denis-Balako — APPROVED** [review]
Good job!

File: src/main/java/mate/academy/dao/impl/CinemaHallDaoImpl.java [hunk 3]
```diff
@@ -20,7 +20,7 @@ public CinemaHall add(CinemaHall cinemaHall) {
         try {
             session = HibernateUtil.getSessionFactory().openSession();
             transaction = session.beginTransaction();
-            session.persist(cinemaHall);
+            session.save(cinemaHall);
```
Comment (nacenik) [inline src/main/java/mate/academy/dao/impl/CinemaHallDaoImpl.java:23]: use `persist`, cause `save` is deprecated for hibernate 6+ version
fix it everywhere

File: src/main/java/mate/academy/dao/impl/ShoppingCartDaoImpl.java [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+@Dao
+public class ShoppingCartDaoImpl implements ShoppingCartDao {
+    @Override
+    public ShoppingCart add(ShoppingCart shoppingCart) {
+        Transaction transaction = null;
+        Session session = null;
+        try {
+            session = HibernateUtil.getSessionFactory().openSession();
+            transaction = session.beginTransaction();
+            session.save(shoppingCart);
```
[the remark this answers is not in the document]
Comment (nacenik) [reply src/main/java/mate/academy/dao/impl/ShoppingCartDaoImpl.java:23]: not fixed 
fixe it everywhere

File: src/main/java/mate/academy/model/User.java [hunk 5]
```diff
@@ -13,7 +12,6 @@ public class User {
     @Id
     @GeneratedValue(strategy = GenerationType.IDENTITY)
     private Long id;
-    @Column(unique = true)
```
Comment (nacenik) [inline src/main/java/mate/academy/model/User.java:16]: email should be unique
2 different users cannot have the same email

**Review by nacenik — CHANGES_REQUESTED** [review]
Good, let's improve it a little bit :)

**Review by vovasalyha — APPROVED** [review]

**Pull request closed without merging**